### PR TITLE
Add drip to destinations

### PR DESCRIPTION
**Pull request by jacobpmeyer**
_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [If destination is already live] Tested for backward compatibility of destination. **Note:** New required fields are a breaking change.
- [ ] [Segmenters] Tested in the staging environment
- [ ] [Segmenters] [If applicable for this change] Tested for regression with Hadron.

### CODE REVIEW

File: packages/destination-actions/src/destinations/drip/trackEvent/index.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      default: {
+        '@path': '$.userId'
+      }
+    }
+  },
+  perform: (request, { settings, payload }) => {
+    // Ensure properties is an object
+    const formattedPayload = {
+      ...payload,
+      properties: typeof payload.properties === 'object' ? payload.properties : {}
```
Comment (stevedrip) [inline packages/destination-actions/src/destinations/drip/trackEvent/index.ts]: properties already has a type constraint of 'object', why is this necessary?

Comment (jacobpmeyer) [reply packages/destination-actions/src/destinations/drip/trackEvent/index.ts]: It wasn't

File: packages/destination-actions/src/destinations/drip/index.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        method: 'get'
+      })
+    }
+  },
+
+  extendRequest({ settings }) {
+    return {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${settings.apiKey}`
```
Comment (stevedrip) [inline packages/destination-actions/src/destinations/drip/index.ts:47]: A user agent would also be nice

Comment (jacobpmeyer) [reply packages/destination-actions/src/destinations/drip/index.ts:47]: Ope just saw this. I'll add it.

File: packages/destination-actions/src/destinations/drip/identify/index.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      default: { '@path': '$.context.ip' }
+    },
+    newEmail: {
+      description: "The person's new email address.",
+      label: 'New Email Address',
+      required: false,
+      type: 'string',
+      format: 'email',
+      default: { '@path': '$.traits.newEmail' }
+    },
```
Comment (stevedrip) [inline packages/destination-actions/src/destinations/drip/identify/index.ts]: I see you added this and I don't disagree. I can't help speculating on what the future might actually look like, since I want to avoid any one-way doors.

I think in the future we might want to pass an id type, and maybe even an id. So `email` would mean change this person's email to X, unless it the id type was `email`.

I think this model could be to far out to consider planning for though 🤔

File: packages/destination-actions/src/destinations/drip/identify/index.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+const person = (payload: Payload) => {
+  return {
+    custom_fields: payload.customFields,
+    email: payload.email,
+    ip_address: payload.ip,
+    new_email: payload.newEmail,
+    sms_number: payload.sms,
+    status: payload.status,
+    status_updated_at: payload.statusUpdatedAt,
+    tags: payload.tags,
```
Comment (stevedrip) [inline packages/destination-actions/src/destinations/drip/identify/index.ts]: This needs to format the tags into an array, right? This is something worth talking to Thomas about. Like, ideally we'd format before displaying to the user as the expected payload.

File: packages/destination-actions/src/destinations/drip/identify/index.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      required: false,
+      type: 'datetime',
+      default: { '@path': '$.traits.statusUpdatedAt' }
+    },
+    tags: {
+      description: "Tags to add to a person's profile.",
+      label: 'Tags',
+      required: false,
+      type: 'string', // comma separated string list of tags e.g. "tag1,tag2"
+      default: { '@path': '$.properties.tags' }
```
Comment (stevedrip) [inline packages/destination-actions/src/destinations/drip/identify/index.ts:78]: Look around this file to see if there is a better option here, such as:
https://github.com/segmentio/action-destinations/blob/main/packages/core/src/mapping-kit/index.ts#L196-L214

File: packages/destination-actions/src/destinations/drip/identify/index.ts (hunk 3 above)
Comment (jacobpmeyer) [reply packages/destination-actions/src/destinations/drip/identify/index.ts]: Are you concerned that without an ID, future events using an old email would get dropped?

I suppose email changes probably aren't a huge slice of customers' event pie, so it might make more sense to not include newEmail and avoid the confusion altogether. 

Would there any worry for added complexity with adding an ID type?

Comment (jacobpmeyer) [reply packages/destination-actions/src/destinations/drip/identify/index.ts]: I just went ahead and removed it. We can always add it in later.

File: packages/destination-actions/src/destinations/drip/identify/index.ts (hunk 4 above)
Comment (jacobpmeyer) [reply packages/destination-actions/src/destinations/drip/identify/index.ts]: Done
